### PR TITLE
Bad input

### PR DESCRIPTION
**Pull request by Cafferoq**
### Summary

Sanitize the inputs server side

### Checklist

- [x] Did you run `npm test`?
- [x] Did you update/add documentation for new methods or changed functionality?

Finally, is it ready to be reviewed and merged: yes :white_check_mark: 

### How to Review

1. Pull this branch
2. On consus-client pull alternate-checkout story branch
3. Search for student with invalid character in student id (non-alphanumeric)
4. Network tab should show failure JSON that states, "Invalid characters in student ID"

NOTE: There is more functionality, but dev is required client side because otherwise the input would be sanitized before it gets to the server.
 

### Links

Put links to Jira tickets, and anything else, here.

- [TFF-90](https://msoese.atlassian.net/browse/TFF-90)

### CODE REVIEW

File: src/api/item.js [hunk 1]
```diff
@@ -5,6 +5,10 @@ import ItemStore from '../store/item-store';
 let app = express();
 
 app.get('/', (req, res) => {
+    let regex = new RegExp("^[a-zA-Z0-9]*$");
```
Comment (mattmahn) [inline src/api/item.js]: Shouldn't it be `^[a-zA-Z0-9]+$` so that it requires at least one character?

File: src/api/student.js [hunk 2]
```diff
@@ -5,8 +5,11 @@ let app = express();
 
 app.get('/', (req, res) => {
     let student = StudentStore.getStudentById(req.query.id);
+    let regex = new RegExp("^[a-zA-Z0-9]*$");
```
Comment (mattmahn) [inline src/api/student.js]: `^[a-zA-Z0-9]+$`?

**Comment by gearedmonkey** [comment]
@Cafferoq These changes prevent student 123456 from popping up on the screen. Is that expected? I have DEV for consus-client and this branch for consus. I pulled DEV of consus and verified that 123456 can be displayed when typing it in, so it seems to occur only in your branch.

File: src/api/item.js (hunk 1 above)
Comment (thofmann) [inline src/api/item.js]: Yes.

File: src/api/student.js (hunk 2 above)
Comment (thofmann) [inline src/api/student.js]: >`^[a-zA-Z0-9]+$`?

Yes.

**Comment by thofmann** [comment]
Coverage is going down since we have nothing in place for writing API tests yet.

I think we can override this check for now, but we should set up tests for the API soon.

**Pull request merged**